wordpress-mobile/WordPress-iOS
# My Site Dashboard: change how API response is parsed and injected into cards

Part of #17873

This is a technical refactor to:

* Use `Decodable` to parse the API response instead of `NSDictionary`
* Inject the entity into the cards
* Adjust naming of a few parameters
* Display only one single cell for posts, which might contain one or more cards

### To test

**Important: make sure to run the API on your sandbox and to have My Site Dashboard flag enabled.**

1. Run the app
2. Tap Dashboard
3. Check that the dashboard is loaded
4. Switch sites a few times

### Instruments: Allocations

1. Run Instruments > Allocations
2. Run the app
3. Start recording
4. Switch blogs (while displaying the dashboard a few times)
5. Compare the number of `PostsCardViewController` and `DashboardPostsCardCell`. The number of `Persistent` for the VC should never be bigger than the number of the `cells * 2` (ideally we will have a maximum of 2, but this will be tackled in a subsequent task)

## Regression Notes
1. Potential unintended areas of impact
n/a

2. What I did to test those areas of impact (or what existing automated tests I relied on)
n/a

3. What automated tests I added (or what prevented me from doing so)
Updated already existent tests.

PR submission checklist:
